docs/kubernetes/user-guides/snapshots.md

```diff
@@ -19,33 +21,102 @@ $ ./deploy/kubernetes/deploy-driver.sh
 
 ### Snapshot Example
 
-1. Create example Default Snapshot Class
+**Create Storage Class:**
+
+If you haven't created a `StorageClass` yet, create one first:
+
+```console
+kubectl apply -f ./examples/kubernetes/snapshot/storageclass.yaml
 ```
-$ kubectl create -f ./examples/kubernetes/demo-defaultsnapshotclass.yaml
+
+For more advance `StorageClass` configuration, please see [Kubernetes Basic User Guide](/docs/kubernetes/user-guides/basic.md).
+
+**Create  Default VolumeSnapshotClass:**
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/default_volumesnapshotclass.yaml
 ```
-2. Create a snapshot of the PVC created in above example
+
+**Create source PVC:**
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/source_pvc.yaml
+```
+
+**Generate sample data:**
+
+Create a sample pod with the source PVC. The source PVC is mounted into `/demo/data` directory of this pod.
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/source_pod.yaml
+```
+
+Now, let's create a file inside `/demo/data` directory:
+
+```console
+kubectl exec source-pod -- touch /demo/data/sample-file.txt
 ```
-$ kubectl create -f ./examples/kubernetes/demo-snapshot.yaml
+
+Check if the file has been created successfully:
+
+```console
+$ kubectl exec source-pod -- ls /demo/data/
+lost+found
+sample-file.txt
 ```
-3. Verify Snapshot is created and is ready to use
+
+**Create a snapshot of the source PVC:**
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/snapshot.yaml
 ```
-$ k get volumesnapshots demo-snapshot-podpvc -o yaml
+
+**Verify Snapshot has been created and it is ready to use:**
+
+```console
+$ kubectl get volumesnapshot snapshot-source-pvc -o yaml
 apiVersion: snapshot.storage.k8s.io/v1alpha1
 kind: VolumeSnapshot
 metadata:
-  creationTimestamp: 2018-10-05T16:59:26Z
-  generation: 1
-  name: demo-snapshot-podpvc
+  ...
+  name: snapshot-source-pvc
   namespace: default
+  resourceVersion: "15485"
+  selfLink: /apis/snapshot.storage.k8s.io/v1alpha1/namespaces/default/volumesnapshots/snapshot-source-pvc
+  uid: f1b8855d-7007-11e9-8f36-42010a800014
+spec:
+  snapshotClassName: default-snapshot-class
+  snapshotContentName: snapcontent-f1b8855d-7007-11e9-8f36-42010a800014
+  source:
+    apiGroup: null
+    kind: PersistentVolumeClaim
+    name: source-pvc
+status:
+  creationTime: "2019-05-06T14:05:04Z"
+  readyToUse: true
+  restoreSize: 1Gi
+```
 
-...
+**Restore the Snapshot into a new PVC:**
 
-status:
-  creationTime: 2018-10-05T16:59:27Z
-  ready: true
-  restoreSize: 6Gi
+Create a new PVC. Specify `spec.dataSource` section to restore from VolumeSnapshot `snapshot-source-pvc`.
 
+```console
+ kubectl create -f ./examples/kubernetes/snapshot/restored_pvc.yaml
 ```
-4. Create a new volume from the snapshot
+
+**Verify sample data has been restored:**
+
+Create a sample pod with the restored PVC:
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/restored_pod.yaml
```

Comment:
do we want to bring down the source pod before bringing up the restored pod or anything?

Reply:
Restore is happening in a different PVC. So we don't need to bring down the source pod.